### PR TITLE
#12203 Fixes VSCode modal inform message focus

### LINKED ISSUE #12203: [VSCode] Modal Information Message doesn't take focus on the first button
### Bug Description:
If I'll open `vscode.window.showInformationMessage("", {modal: true} , "I'm focused!", "Secondary")` , the focus will be on the background, but in VSCode it will be on the button "I'm focused!".

### Steps to Reproduce:

1. Get example VSCode extension [modalinformmessage-0.0.1.zip](https://github.com/eclipse-theia/theia/files/10802107/modalinformmessage-0.0.1.zip) 
2. Install it in VSCode
3. Call the comand "Focus Test"
4. It will open dialog and set focus on "I'm focused!" 
5. If we press enter, notification "Button was onFocus" will pop up.
6. Start Theia and install the same extension in there
7. Call the comand "Focus Test"
8. It will open dialog and focus will be left where it was before
9. If you press "Enter", the dialog will not close and notification will not pop up (depends on where last focus was)

### Additional Information

- Operating System: Windows 
- Theia Version: 1.24 and master

### PR DESCRIPTION
**Pull request by Zebsterpasha**
Fixes VSCode modal information message focus.
Now focus will be on the first button, not on a background as it was before. 

Closes #12203
Closes https://github.com/eclipse-theia/theia/pull/12204
Signed-off-by: Pavel Nikolaev <pnik@1c.ru>

#### What it does

Added a condition, if there is a first button on a Modal Message, it will be created by appendAcceptButton, other buttons will be created with createButton, except "Cancel".

#### How to test

1. Start Theia and install the extension [modalinformmessage-0.0.1.zip](https://github.com/eclipse-theia/theia/files/10803197/modalinformmessage-0.0.1.zip)
2. Call the comand "Focus Test"
3. It will open the dialog and focus will be on the first button (onot on a background as it was before)
4. If you press "Enter", the dialog will close and notification "Button was onFocus" will pop up.

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/plugin-ext/src/main/browser/dialogs/modal-notification.ts [hunk 1]
```diff
@@ -12,6 +12,9 @@
 // https://www.gnu.org/software/classpath/license.html.
 //
 // SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+//
+// Contributors:
+//      Pavel Nikolaev <pnik@1c.ru> (1C-Soft LLC) - Issue #12203
```
Comment (msujew) [inline packages/plugin-ext/src/main/browser/dialogs/modal-notification.ts:17]: We don't perform personal committer attributions in this project. Please don't modify the header unless you created the file yourself.

File: packages/plugin-ext/src/main/browser/dialogs/modal-notification.ts [hunk 2]
```diff
@@ -77,8 +80,13 @@ export class ModalNotification extends AbstractDialog<string | undefined> {
...
 
-        actions.forEach((action: MainMessageItem) => {
-            const button = this.createButton(action.title);
+        actions.forEach((action: MainMessageItem, index: number) => {
+            let button;
+            if (index === 0) {
+                button = this.appendAcceptButton(action.title);
+            } else {
+                button = this.createButton(action.title);
+            }
```
Comment (msujew) [inline packages/plugin-ext/src/main/browser/dialogs/modal-notification.ts:89]: Suggestion:

```suggestion
        actions.forEach((action: MainMessageItem, index: number) => {
            const button = index === 0
                ? this.appendAcceptButton(action.title)
                : this.createButton(action.title);
```

**Review by msujew — COMMENTED** [review]
Hey @Zebsterpasha, thank you for your contribution :)

File: packages/plugin-ext/src/main/browser/dialogs/modal-notification.ts (hunk 1 above)
Comment (Zebsterpasha) [reply packages/plugin-ext/src/main/browser/dialogs/modal-notification.ts:17]: Okey, thank you for the advice)

**Review by msujew — APPROVED** [review]
Thanks, looks good to me 👍

**Pull request merged**